lib/nativeTimerifyWrap.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+const { performance, PerformanceObserver } = require('perf_hooks');
+
+// TODO: use WeakRef and finalizationRegistry to trigger the cleanup?
+const fMap = new Map();
+let obs = null;
+
+module.exports = function nativeTimerifyWrap(name, fn, onSend, opts) {
+    const wrapped = performance.timerify(fn, opts);
+    return function timerified(...args) {
+        if (!obs) {
+            obs = new PerformanceObserver((list) => {
+                for (const entry of list.getEntries()) {
+                    const name = fMap.get(entry.name);
+                    /* istanbul ignore else */
+                    if (name) {
+                        fMap.delete(entry.name);
+                        onSend(name, entry);
+                    }
+                    if (fMap.size === 0) {
+                        obs.disconnect();
+                        obs = null;
+                    }
+                }
+            });
+            obs.observe({ entryTypes: ['function'] });
+        }
+        fMap.set(fn.name, name);
```

Comment:
We're thinking with Anto that we can wrap the original function with another with a random name is an arrow, so that we avoid clashing, we'll talk about that tomorrow in dept

Reply:
I would leave things as they are for now. I think the fix would be easier in Node core than here.

Reply:
lgtm, I'm just not so happy to ship something that may give bad results if used with arrows, I know that we can't do nothing about it, it's an itch of mine.

Reply:
Okey 👍